nomt/src/beatree/writeout.rs

```diff
@@ -74,34 +71,25 @@ fn do_run(mut cx: Cx, mut io: IoDmux) {
     cx.bbn_write_out.wait_writes(&mut io);
     cx.ln_write_out.wait_writes(&mut io);
 
-    cx.bbn_write_out.fsync(&mut io);
-    cx.ln_write_out.fsync(&mut io);
+    cx.bbn_write_out.send_fsync(&mut io);
+    cx.ln_write_out.send_fsync(&mut io);
 
-    loop {
-        // At this point, the BBN and LN files are fully synced. We can now update the meta file.
-        if let Some(ref mut meta_swap) = &mut cx.meta_swap {
-            if meta_swap.run(&mut io) {
-                cx.meta_swap = None;
-                return;
-            }
-        }
-    }
+    cx.bbn_write_out.wait_fsync(&mut io);
+    cx.ln_write_out.wait_fsync(&mut io);
```

Comment:
seems better to dispatch the fsyncs at the same time and then handle the completions async.

for a follow-up.